src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,266 @@
+            if (author.getEmailAddress() != email) {
+                continue
+            }
+            line.printme()
+            println("  Age: ${line.age} secs")
+            sum += line.age
+            total++
+        }
+
+        //println("All lines:")
+        //codeLines.forEach { line -> line.printme() }
+
+        var avg = if (total > 0) sum / total else 0
+        println("avg code line age for <$email> is ${avg} seconds, lines total: ${total}")
+    }
+
+    /**
+     * Scans through the repo for alive and deleted code lines, and stores them
+     * in the |codeLines| list.
+     */
+    private fun compute() {
+        val treeWalk = TreeWalk(repo)
+        treeWalk.setRecursive(true)
+        treeWalk.addTree(head.getTree())
+
+        val files: MutableMap<String, ArrayList<RevCommitLine>> = mutableMapOf()
+
+        // Build a map of file names and their code lines.
+        while (treeWalk.next()) {
+            val path = treeWalk.getPathString()
+            val lines = getFileLines(treeWalk.getObjectId(0), path, head)
+            if (lines != null) {
+                files.put(path, lines)
+            }
+        }
+  
+        val df = DiffFormatter(DisabledOutputStream.INSTANCE)
+        df.setRepository(repo)
+        df.setDetectRenames(true)
+
+        val revWalk = RevWalk(repo)
+        revWalk.markStart(head)
+
+        var commit: RevCommit? = revWalk.next()  // move the walker to the head
+        do {
+            var parentCommit: RevCommit? = revWalk.next()
+
+            println("commit: ${commit!!.getName()}; '${commit.getShortMessage()}'")
+            if (parentCommit != null) {
+                println("parent commit: ${parentCommit.getName()}; '${parentCommit.getShortMessage()}'")
+            }
+            else {
+                println("parent commit: null")
+            }
+
+            // A step back in commits history. Update the files map according
+            // to the diff.
+            val diffs = df.scan(parentCommit, commit)
+            for (diff in diffs) {
+                val oldPath = diff.getOldPath()
+                val oldId = diff.getOldId().toObjectId()
+                val newPath = diff.getNewPath()
+                val newId = diff.getNewId().toObjectId()
+                println("old: '$oldPath', new: '$newPath'")
+
+                // Skip binary files.
+                var fileId =
+                    if (newPath != DiffEntry.DEV_NULL) newId else oldId
+                if (RawText.isBinary(repo.open(fileId).openStream())) {
+                    continue
+                }
+
+                // TODO(alex): does it happen in the wilds?
+                if (diff.changeType == DiffEntry.ChangeType.COPY) {
+                    continue
+                }
+
+                // File was deleted, put its lines into the files map.
+                if (diff.changeType == DiffEntry.ChangeType.DELETE) {
+                    var lines = getFileLines(oldId, oldPath, commit)!!
```

Comment:
That shouldn't happen because we filter out binary files above so I use !! to terminate the program since something bad and unpredictable happening. Is NPE more suitable for this kind of things?